### PR TITLE
Import bytebuddy implementation from Dubbo 3.2

### PR DESCRIPTION
**Pull request by finefuture**
## What is the purpose of the change

ref: [[Task] Introduce bytebuddy proxy in apache/dubbo-spi-extensions](https://github.com/apache/dubbo/issues/13867)

## Brief changelog

XXXXX

## Verifying this change

XXXXX

Follow this checklist to help us incorporate your contribution quickly and easily:

- [x] Make sure there is a [GITHUB_issue](https://github.com/apache/dubbo/issues) field for the change (usually before
  you start working on it). Trivial changes like typos do not require a GITHUB issue. Your pull request should address
  just this issue, without pulling in other changes - one PR resolves one issue.
- [ ] Format the pull request title like `[Dubbo-XXX] Fix UnknownException when host config not exist #XXX`. Each commit
  in the pull request should have a meaningful subject line and body.
- [ ] Write a pull request description that is detailed enough to understand what the pull request does, how, and why.
- [ ] Write necessary unit-test to verify your logic correction, more mock a little better when cross module dependency
  exist. If the new feature or significant change is committed, please remember to add sample
  in [dubbo samples](https://github.com/apache/dubbo-samples) project.
- [ ] Run `mvn clean install -DskipTests=false` & `mvn clean test-compile failsafe:integration-test` to make sure
  unit-test and integration-test pass.
- [ ] If this contribution is large, please follow
  the [Software Donation Guide](https://github.com/apache/dubbo/wiki/Software-donation-guide).

### CODE REVIEW

**Comment by finefuture** [comment]
@AlbumenJ @songxiaosheng PTAL

File: dubbo-proxy-extensions/dubbo-proxy-bytebuddy/pom.xml [hunk 1]
```diff
@@ -0,0 +1,38 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (AlbumenJ) [inline dubbo-proxy-extensions/dubbo-proxy-bytebuddy/pom.xml:1]: Please add ASF license header

File: dubbo-proxy-extensions/pom.xml [hunk 2]
```diff
@@ -0,0 +1,20 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<project xmlns="http://maven.apache.org/POM/4.0.0"
```
Comment (AlbumenJ) [inline dubbo-proxy-extensions/pom.xml:2]: Add for this one too

**Review by AlbumenJ — CHANGES_REQUESTED** [review]

File: dubbo-proxy-extensions/dubbo-proxy-bytebuddy/pom.xml (hunk 1 above)
Comment (finefuture) [reply dubbo-proxy-extensions/dubbo-proxy-bytebuddy/pom.xml:1]: Okay.

File: dubbo-proxy-extensions/pom.xml (hunk 2 above)
Comment (finefuture) [reply dubbo-proxy-extensions/pom.xml:2]: Okay.

**Review by AlbumenJ — APPROVED** [review]

**Pull request merged**